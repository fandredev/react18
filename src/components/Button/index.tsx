import React, {ReactNode} from 'react'
import style from "./button.module.scss"

interface ButtonProps {
  children: ReactNode
  type?: 'submit' | 'button' | 'reset' | undefined
  onClick?: () => void
}


export default function Button({onClick, type, children}: ButtonProps) {
  return (
    <button onClick={onClick} type={type} className={style.button}>{children}</button>
  )
}

// export default class Button extends React.Component<ButtonProps> {
//
//   render(){
//     const { children, type = 'button', onClick } = this.props
//     return (
//       <button onClick={onClick} type={type} className={style.button}>{children}</button>
//     )
//   }
// }
