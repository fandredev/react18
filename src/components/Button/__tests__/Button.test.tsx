import { render, screen } from "@testing-library/react";
import Button from "../index";

describe(`Test component: ${Button.name}`, () => {

  test('Should be possible to snapshot the Button component when the page is rendered', () => {
    const { container } = render(<Button type="button">Começar</Button>)

    expect(container).toMatchSnapshot()
  })

  test('Should be able to load the button component with the children props', () => {
    render(<Button type="button">Começar</Button>)

    const button = screen.getByRole('button')
    expect(button.textContent).toBe('Começar')
  })
})
