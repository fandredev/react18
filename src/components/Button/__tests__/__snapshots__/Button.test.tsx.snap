// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component: Button Should be possible to snapshot the Button component when the page is rendered 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    Come√ßar
  </button>
</div>
`;
